Add unloadability support to ReliabilityFramework

janvorli opened the pull request:
This change adds support for unloadable AssemblyLoadContext to the GC
ReliabilityFramework. It basically mimics what was there ifdef-ed out
for AppDomains.
It allows stress testing GC when running inside of an unloadable
AssemblyLoadContext. GC has some special handling for collectible
classes and this allows testing the respective code paths.

Maoni0 commented on tests/src/GC/Stress/Framework/LoaderClass.cs:
```diff
@@ -126,6 +126,102 @@ public void Load(string assemblyName, string[] paths, ReliabilityFramework rf)
+    /// </summary>
+    /// <param name="path">The assembly to load</param>
+    /// <param name="paths">Paths to search for the given assembly</param>
+    public void LoadFrom(string path, string[] paths, object rf)
```
why object as the type here instead of ReliabilityFramework?

Maoni0 commented on tests/src/GC/Stress/Framework/LoaderClass.cs:
```diff
@@ -126,6 +126,102 @@ public void Load(string assemblyName, string[] paths, ReliabilityFramework rf)
+    /// <param name="path">The assembly to load</param>
+    /// <param name="paths">Paths to search for the given assembly</param>
+    public void LoadFrom(string path, string[] paths, object rf)
+    {
```
so I see there's a lot of duplicated code and I am not asking you to spend more time to not duplicate it. but would you mind adding a one line comment just to say what the difference is between this and AD for each of the duplicated code block?

janvorli replied:
I wonder, should we instead delete all the AppDomain related stuff from the framework? I have found that it has already bitrotted at several places.

janvorli commented:
@Maoni0 I've merged most of the differences between AppDomain and AssemblyLoadContext stuff and added a comment to a function where the merge would not make sense. 
I have also modified the way we run the tests in AssemblyLoadContexts so that we now use a separate thread per each tests like the AppDomain version was doing. I have overlooked this during the initial porting.
